author/task.py

```diff
@@ -562,12 +550,45 @@ def _get_commandlist_with_script_call(self, task):
             classname = re.sub(r"Overriden$", "", self.__class__.__name__)
         else:
             classname = self.__class__.__name__
+
         script = "from jobtronaut.author.plugins import Plugins;" \
-                 "Plugins().task(\"{0}\")(\"{1}\").script()".format(classname, arguments)
+                 "task=Plugins().task(\"{0}\")(\"{1}\");task.script()".format(classname, arguments)
+
+        script = script + ";task.neutralize_commands()" if task.no_retry else script
+
         cmd = self._get_commandlist_with_resolved_executable(task)
         cmd.append(script)
+
         return cmd
 
+    @staticmethod
+    def neutralize_commands():
+        """ neutralize all commands of the current task
+
+        """
+        from jobtronaut.query import (
+            initialize_engine,
+            tractor_query
+        )
+        initialize_engine()
+
+        # if a command runs he normally has access to the `TR_ENV_* vars,
+        # so we know where we want to neutralise all commands
+        job_id = os.getenv("TR_ENV_JID")
+        task_id = os.getenv("TR_ENV_TID")
+
+        if not job_id and task_id:
```

Comment:
```suggestion
        if not (job_id and task_id):
```
The parenthesis are important due to the operator precedence.

Reply:
Indeed.

Reply:
Indeed.